Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -413,102 +419,94 @@ namespace AzToolsFramework
             return false;
         }
 
-        // if the field is not empty then we have a file linked to the prefab.
+        // record an undo step if a local view bookmark component is added and configured
+        ScopedUndoBatch undoBatch("SetupLocalViewBookmarks");
+        undoBatch.MarkEntityDirty(bookmarkComponent->GetEntityId());
+
+        // initialize default locations to 0. This is a temporary solution to match the 12 locations of the legacy system
+        // once there is a UI for the view bookmarks these lines should be removed
+        auto createDefaultBookmarksFn = [this, &registry]()
+        {
+            if (m_localBookmarks.empty())
+            {
+                for (int i = 0; i < DefaultViewBookmarkCount; i++)
+                {
+                    AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
```

Comment:
`finalPath` can be const.

nit: Also, only the last + operation is needed inside the loop, all the others can be cached outside.

Reply:
This has now been updated 👍